### DISCUSSION TITLE
button click event causes xamal not to display properly

### DISCUSSION THREAD

**Discussion by zy1075984** [AvaloniaUI/Avalonia#13205] [Q&A]
`<Button Grid.Row="4" Grid.Column="1" Name="btnSetIP" Height="30" Width="60" HorizontalAlignment="Right" Click="btnSetIP_ClickHandler"/>`

axaml error:
Unable to find suitable setter or adder for property Click of type Avalonia.Controls:Avalonia.Controls.Button for argument System.Private.CoreLib:System.String, available setter parameter lists are:
System.EventHandler`1[[Avalonia.Interactivity.RoutedEventArgs, Avalonia.Base, Version=11.0.2.0, Culture=neutral, PublicKeyToken=c8d484a7012f9a8b]] Line 50, position 111.	FLuX.IoT.DrvAvCrossDef	D:\zhangyi\Programe\FLuX.IoT\FLuX.IoT.DrvInf\FLuX.IoT.DrvAvCrossDef\Views\UCColGatewaySettingView.axaml	50

**Comment by timunie** [comment]
show us the click handler. Most likely EventArgs are wrong

**Reply by zy1075984** [reply]
public void btnSetIP_ClickHandler(object sender, RoutedEventArgs args)

**Reply by timunie** [reply]
does it work for you after project build now? 

Note: The event must be placed in the `MyWindow.axaml.cs` file. Not inside your ViewModel.

**Comment by maxkatz6** [comment]
Your projects needs to be rebuild after adding event handler.
